onnxruntime/test/providers/provider_test_utils.h

```diff
@@ -307,6 +323,23 @@ class OpTester {
     AddSeqData<T>(output_data_, name, seq_tensors);
   }
 
+  template <typename T>
+  void AddOptionalTypeTensorInput(const char* name, const std::vector<int64_t>& dims,
+                                  const std::initializer_list<T>* values = nullptr,
```

Comment:
Probably needs some doco here explaining the difference between an optional input that's missing (AddMissingOptionalInput), and an input that's an 'optional', as that's a lot of intrinsic knowledge required to understand the different methods.

Reply:
Changed some of the nomenclature around `AddMissingOptional...` in https://github.com/microsoft/onnxruntime/pull/8292